### DISCUSSION TITLE
Filter data using IN specific to user

### DISCUSSION THREAD

**Discussion by applicationuser** [Budibase/budibase#5407] [Help]
Hello Budibase Community,

I have two tables’ Products and Product Data, logged in user has access to either single or multiple or all products in the table.

To supply user mapping to product, created relationship between users and product table. Assigning specific products to user.
```
Product -> Product Name, Region, Users (BD user)
Product Data -> Product Name, Month, Value (etc. other fields) 
```
Requirement is to show table Product data information to user based on his products access. 

There will be multiple products a user can access.

how can we filter data from Product Data table based on user products? Define filter does not have choice to use IN operator.

i search in community but did not find any relevant article that helps, please point me to that discussion if it answered.

Thank you for your time

**Comment by melohagan** [comment]
Hey @applicationuser 

Is there also a relationship between your Product and Product Data tables? 

In that case you wouldn't need a filter, because you'd be getting the product data for each product that the user has.

**Reply by applicationuser** [reply]
Hello @melohagan 

Thanks for your reply,

Yes there is relationship between those tables, i have added it now.  

my page design looks like below,  if we set User Data Provider Limit > 1 then table is repeated for each user product.   

would it possible to display all of the products data in single table for better user experience, 

![image](https://user-images.githubusercontent.com/42922833/163410812-56bb969d-ba8f-4bb1-baf7-e34ce5819008.png)

**Comment by melohagan** [comment]
Hey @applicationuser 

The easiest solution here would be to have a data provider pointing to the current users products:
<img width="233" alt="Screenshot 2022-04-14 at 22 36 10" src="https://user-images.githubusercontent.com/101575380/163480798-b11a2614-b803-4d08-baf9-e1a39a36c1ec.png">

Then you just need to use formula columns on the product table to show product data fields, as outlined here: https://docs.budibase.com/docs/formula#fields-from-relationships

**Reply by applicationuser** [reply]
Hello @melohagan 

Thanks for your reply, i looked at the formula field , that would not work as we have multiple rows of product data for each product. its one to many.   

Its dynamic more data can be added to product data table in future.

**Reply by melohagan** [reply]
You could try some of the aggregation handle bars. For example: `{{ pluck Product Data 'Month' }}` will list all of the months as a single value:

<img width="254" alt="Screenshot 2022-04-15 at 15 50 12" src="https://user-images.githubusercontent.com/101575380/163585366-db636c17-909a-4401-9276-f57d8ad79735.png">

If you scroll down the left-hand side of the Formula dialog, you can see the list of helpers:
 
<img width="340" alt="Screenshot 2022-04-15 at 15 50 49" src="https://user-images.githubusercontent.com/101575380/163585437-65c5e922-a3cb-4619-87c7-ebbf1d8b9cc0.png">
